app/components/edit/EditAddress.tsx

```diff
@@ -1,31 +1,62 @@
 import React, { useState } from "react";
 import ReactModal from "react-modal";
-import PropTypes from "prop-types";
 import {
   RiDeleteBin5Line,
   RiEditBoxLine,
   RiArrowLeftLine,
 } from "react-icons/ri";
+import type { InternalAddress } from "../../pages/OrganizationEditPage";
+import type { Address } from "../../models";
 
 import s from "./EditAddress.module.scss";
 
 // Subcomponents
 
+// Note: This must be manually updated to match the actual <input> fields in the
+// EditAddressModal component below.
+interface EditAddressFormData {
+  name: string;
+  address_1: string;
+  address_2: string;
+  city: string;
+  state_province: string;
+  postal_code: string;
+}
+
+type EditAddressModalProps = {
+  isOpen: boolean;
+  onRequestClose: () => void;
+  defaultData: InternalAddress | Record<string, never>;
+  onSave: (address: InternalAddress) => void;
+};
+
 /** Modal containing the edit and add new address form.
  *
  * The form data is only synchronized with the outer component's state when the
  * form is submitted. This allows the user to back out of any changes with the
  * "Cancel" button. Therefore, we have chosen to implement the form inputs as
  * uncontrolled components.
  */
-const EditAddressModal = ({ isOpen, onRequestClose, defaultData, onSave }) => {
-  const isEdit = !!defaultData.id;
+const EditAddressModal = ({
+  isOpen,
+  onRequestClose,
+  defaultData,
+  onSave,
+}: EditAddressModalProps) => {
+  const isEdit = "id" in defaultData && !!defaultData.id;
   const title = isEdit ? "Edit Address" : "Add New Address";
   const submitButtonText = isEdit ? "Save Address" : "Add New Address";
-  const handleSubmit = (e) => {
+  const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
     e.preventDefault();
-    const formData: any = new FormData(e.currentTarget);
-    const newData = Object.fromEntries(formData.entries());
+    const formData = new FormData(e.currentTarget);
+    // We need to perform a double type assertion with the intermediate `as
+    // unknown` because the types for FormData.entries() don't exactly line up
+    // with EditAddressFormData. FormData.entries() allows values to either be
+    // strings or files, but EditAddressFormData's values can only be strings.
+    // Our form doesn't include a file upload, so we cast it away.
+    const newData: EditAddressFormData & { id?: number } = Object.fromEntries(
+      formData.entries()
+    ) as unknown as EditAddressFormData;
```

Comment:
Wow, didn't know you could do this!

Reply:
Yeah, this got a bit more complicated than I really wanted it to be, but I didn't want to have to have to define and name a new type, so I went with creating an anonymous type intersection.

As cool as it is to be able to do this kind of stuff, I generally think that if we need complicated types like this, then the design of our program isn't exactly right. Hoping to tackle actual refactoring soon, after this TypeScript migration is done!